File: go/fibonacci/fibonacci.go
package main

import "fmt"

func main() {
    fmt.Println(fibonacciLine(10))
}

func fibonacci(n int) int {
    if n == 0 || n == 1 {
        return n
    }
    return fibonacci(n-1) + fibonacci(n-2)
}

func fibonacciLine(n int) []int {
    var list []int

    for x := 0; x < n; x++ {
        list = append(list, fibonacci(x))
    }

    return list
}

File: dart/fibonacci/fibonacci.dart

main () {
    print(fibonacciLine(10));
}

num fibonacci(num n) {
    if (n == 0 || n == 1) {
        return n;
    } else {
        return fibonacci(n-1) + fibonacci(n-2);
    }
}

List fibonacciLine(num n) {
    var list = new List(n);
    for (var x = 0; x < n; x++) {
        list[x] = fibonacci(x);
    }
    return list;
}

